### PR TITLE
add new generalized & refactored publish-draft-services.py script 

### PR DESCRIPTION
**Pull request by risicle**
~A draft because it's still missing tests, but script seems to work from my ad-hoc testing so take a look.~

https://trello.com/c/V2miZJ5G

Should be ready to review now, has a _reasonable_ level of testing - a few corner cases missed out though.

### CODE REVIEW

File: dmscripts/publish_draft_services.py [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+        service_id = random.randint(55500000, 55599999)
+        get_logger().info(
+            "supplier %s: draft %s: dry run: generating random test service id: %s",
+            draft_service["supplierId"],
+            draft_service['id'],
+            service_id,
+        )
+    else:
+        try:
+            services = client.publish_draft_service(draft_service['id'], 'publish_draft_services.py')
```
Comment (katstevens) [inline dmscripts/publish_draft_services.py]: Should we make it clearer that `publish_draft_services.py` is the `updated_by` user? I know it's not a named kwarg on the client so maybe a comment, or sticking this in a suitably named variable...

File: dmscripts/publish_draft_services.py [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+        documents_bucket.copy(
+            src_bucket=draft_bucket.bucket_name,
+            src_key=draft_document_path,
+            target_key=live_document_path,
+            acl='public-read',
+        )
+
+
+# this function exposed as importable - the intention is that a user of this module will wrap up a custom version
+# of this function and supply that to publish_draft_services as copy_documents_callable
```
[the remark this answers is not in the document]
Comment (risicle) [reply dmscripts/publish_draft_services.py]: In reality, probably adapted for the framework, but it's designed to be opaque to what it's being customized for.

File: scripts/publish-draft-services.py [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+from dmscripts.helpers.auth_helpers import get_auth_token
+from dmscripts.helpers.logging_helpers import configure_logger
+from dmscripts.publish_draft_services import publish_draft_services, copy_draft_documents
+
+_document_keys_by_framework_family = {
+    "g-cloud": (
+        'pricingDocumentURL',
+        'serviceDefinitionDocumentURL',
+        'sfiaRateDocumentURL',
+        'termsAndConditionsDocumentURL',
```
[the remark this answers is not in the document]
Comment (risicle) [reply scripts/publish-draft-services.py]: But ... to where? And how will they be served?

[the remark this answers is not in the document]
Comment (katstevens) [reply scripts/publish-draft-services.py]: Exactly... needs clarification from product before we do anything

File: dmscripts/publish_draft_services.py (hunk 1 above)
Comment (risicle) [reply dmscripts/publish_draft_services.py]: Sure.

Comment (risicle) [reply dmscripts/publish_draft_services.py]: ...but... hm... really I feel it should give space to put the *actual user* here which the script should probably take as an argument. But if it's being run from jenkins...? In theory we could try and extract the jenkins user name of the person that hit the button...

File: scripts/publish-draft-services.py [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+    STAGE = arguments['<stage>']
+    api_url = get_api_endpoint_from_stage(STAGE)
+
+    client = dmapiclient.DataAPIClient(api_url, get_auth_token('api', STAGE))
+    DRAFT_BUCKET = arguments.get('<draft_bucket>') and S3(arguments['<draft_bucket>'])
+    DOCUMENTS_BUCKET = arguments.get('<documents_bucket>') and S3(arguments['<documents_bucket>'])
+    DRY_RUN = arguments['--dry-run']
+    FRAMEWORK_SLUG = arguments['<framework_slug>']
+    DRAFT_IDS_PATH = arguments['--draft-ids']
+    SKIP_DOCS_IF_PUBLISHED = arguments['--skip-docs-if-published']
```
[the remark this answers is not in the document]
Comment (risicle) [reply scripts/publish-draft-services.py]: We've got to leave *some* of the soul of kev in here...

File: scripts/publish-draft-services.py [hunk 5]
```diff
@@ -0,0 +1,98 @@
+#!/usr/bin/env python3
```
[the remark this answers is not in the document]
Comment (risicle) [reply scripts/publish-draft-services.py]: Done

File: tests/test_publish_draft_services.py [hunk 6]
```diff
@@ -0,0 +1,271 @@
...
+        mock_data_api_client,
+        "g-cloud-123",
+        DraftServiceStub(
+            id=379001,
+            framework_slug="g-cloud-123",
+            supplier_id="171400",
+            service_id="2229090909",
+            **{
+                "chinChopperURL": "https://gin.ger.br/ead/g-cloud-123/submissions/171400/379001-steak.pdf",
+                "beeoTeeTomURL": "https://gin.ger.br/ead/g-cloud-123/submissions/171400/379001-kidney.pdf",
```
Comment (katstevens) [inline tests/test_publish_draft_services.py]: So this one should raise the ValueError, correct? Do we assert that?

Comment (risicle) [reply tests/test_publish_draft_services.py]: Ooh well spotted - I realized this one was a tricky situation because the only side effect it has is a possible log message going out. _But_ the thing we really care about the script's behaviour in handling this exception is that it doesn't cause it to abort the run. So in essence the fact that the script continues and checks the following drafts _is_ the assertion.

Comment (katstevens) [reply tests/test_publish_draft_services.py]: Ah I see - we raise if it's anything other than 'file is already here'. Cool 👍
